PreLoad.cmake

```diff
@@ -64,7 +64,9 @@ execute_process(COMMAND uname -m OUTPUT_VARIABLE ARCH)
 if (OS MATCHES "Linux"
     # some build systems may use CC/CXX env variables
     AND "$ENV{CC}" STREQUAL ""
-    AND "$ENV{CXX}" STREQUAL "")
+    AND "$ENV{CXX}" STREQUAL ""
+    AND NOT DEFINED CMAKE_C_COMPILER
+    AND NOT DEFINED CMAKE_CXX_COMPILER)
```

Comment:
(not really related to this PR: I tried it out the previous code state (left side) on my system using `rm -rf build && cmake -S . -B build -DCMAKE_CXX_COMPILER=foo` and env variables `$CC` and `$CXX` being unset. Looking at the PR description, one would expect that cache variable `CMAKE_CXX_COMPILER` will be "clang++". After l. 77 (which is executed), this is however not the case, cache variable `CMAKE_CXX_COMPILER` is "foo". So apparently, the forced write in l.77 did not work, at least not on my system. More weirdly, running it twice

```
set(CMAKE_CXX_COMPILER "clang++" CACHE INTERNAL "")
set(CMAKE_CXX_COMPILER "clang++" CACHE INTERNAL "")
```
sets cache variable `CMAKE_CXX_COMPILER` to clang++ as expected. But that's all unrelated to your PR of course.)

Reply:
>one would expect that cache variable CMAKE_CXX_COMPILER will be "clang++".

Why it should be executed if you set `CMAKE_CXX_COMPILER=foo`?
Am I missing something?

Reply:
Ah, now I see that your PR is about repeated invocations of CMake. When I tried to reproduce the problem, I only ran CMake and started compilation afterwards in which case the passed `-DCMAKE_C(XX)_COMPILER` was correctly propagated.

But in the meantime I could reproduce the problem and can confirm that the PR fixes it.